### PR TITLE
Consuming new Maestro task

### PR DESCRIPTION
**Pull request by jcagme**
Part two of: https://github.com/dotnet/arcade/issues/5693

Consume the new Maestro Task package which knows how to deal with signing metadata when it is present in the asset manifests. The `MergedManifest.xml` will only be pushed as a blob artifact when the asset manifests have a PublishingVersion >= 3.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/PublishBuildAssets.proj [hunk 1]
```diff
@@ -28,10 +37,36 @@
...
+			<!-- When we have <account>.visualstudio.com -->
+      <AzureDevOpsAccount Condition="$(CollectionUri.IndexOf('visualstudio.com')) >= 0">$(CollectionUri.Split('.')[0].Split('/')[2])</AzureDevOpsAccount>
+    </PropertyGroup>
+
+    <!--
+      The new Maestro/BAR build model keeps separate Azure DevOps and GitHub build information.
+      The GitHub information will be extracted based on the Azure DevOps repository.
+    -->
+    <ItemGroup>
+      <ManifestBuildData Include="InitialAssetsLocation=$(SYSTEM_TEAMFOUNDATIONCOLLECTIONURI)$(SYSTEM_TEAMPROJECT)/_apis/build/builds/$(BUILD_BUILDID)/artifacts" />
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/PublishBuildAssets.proj:55]: I thought these were being merged from the input manifests?

Comment (jcagme) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/PublishBuildAssets.proj:55]: We are only using the PublishingVersion but now that you mention it, it makes sense to use these metadata from the assets

Comment (mmitche) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/PublishBuildAssets.proj:55]: Yeah, simplifies code here and would basically all be the same thing anyway.

Comment (jcagme) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/PublishBuildAssets.proj]: Fixed. Test build here https://dnceng.visualstudio.com/internal/_build/results?buildId=725397&view=results

**Pull request merged**